### PR TITLE
feat(tdp-4493): add modulo action

### PR DESCRIPTION
**Pull request by tlnd-cgall**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-4493

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1076/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1076/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1082/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1082/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1083/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1083/)

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    /**
+     * Check that the selected column parameter is correct : defined in the parameters and there's a matching column. If
+     * the parameter is invalid, an exception is thrown.
+     *
+     * @param parameters where to look the parameter value.
+     * @param rowMetadata the row where to look for the column.
+     */
+    private void checkParameters(Map<String, String> parameters, RowMetadata rowMetadata) {
+        if (parameters.get(MODE_PARAMETER).equals(CONSTANT_MODE) && !parameters.containsKey(DIVISOR)) {
+            throw new TalendRuntimeException(ActionErrorCodes.BAD_ACTION_PARAMETER,
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java]: Try to not send an exception (because this is resource consuming)  if the parameters are wrong, just cancel the action like it is made here : 
https://github.com/Talend/data-prep/blob/master/dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/fill/GenerateSequence.java#L98-L98
or like you are doing in compile method and log the problem

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        final String newColumnId = context.column("result");
+
+        if (parameters.get(MODE_PARAMETER).equals(CONSTANT_MODE)) {
+            divisor = parameters.get(DIVISOR);
+        } else {
+            final ColumnMetadata selectedColumn = rowMetadata.getById(parameters.get(SELECTED_COLUMN_PARAMETER));
+            divisor = row.get(selectedColumn.getId());
+        }
+
+        if (NumericHelper.isBigDecimal(value) && NumericHelper.isBigDecimal(divisor)) {
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java]: Maybe you could move this action in the check parameter part too, instead of here, in order to do the check only once and not for all the cells. This values will remain the same during the action.

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
+package org.talend.dataprep.transformation.actions.math;
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java]: The code is not formatted in the Talend Way (indentation, coyright,...)

File: dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+@Action(AbstractActionMetadata.ACTION_BEAN_PREFIX + Modulo.MODULO_NAME)
+public class Modulo extends AbstractActionMetadata implements ColumnAction, OtherColumnParameters {
+
+    protected static final String MODULO_NAME = "modulo";
+
+    /**
+     * Mode: tells if operand is taken from another column or is a constant
+     */
+    public static final String MODE_PARAMETER = "mode";
```
Comment (smallet) [inline dataprep-actions/src/main/java/org/talend/dataprep/transformation/actions/math/Modulo.java]: Use the one defined in OtherColumnParameters.
Remove this line is enough.

File: dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages.properties [hunk 5]
```diff
@@ -623,6 +623,18 @@ parameter.mode.desc=Select if you want to use a value or another column
 parameter.operand.label=Operand
 parameter.operand.desc=Fixed value to use as operand
 
+#
+# Modulo
+#
+action.modulo.label=Modulo
+action.modulo.desc=Perform modulo operation on this column with another one or with a fixed value
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages.properties]: The description seems to be not the same as in th JIRA ticket : "Computing the resulting modulo of a division". So the same in the french key.

File: dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages.properties [hunk 6]
```diff
@@ -623,6 +623,18 @@ parameter.mode.desc=Select if you want to use a value or another column
...
+
+parameter.modeModulo.label=Use with
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages.properties]: The label seems to be not the same as in th JIRA ticket.  So the same in the french key.

File: dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/math/ModuloTest.java [hunk 7]
```diff
@@ -0,0 +1,219 @@
+package org.talend.dataprep.transformation.actions.math;
```
Comment (tlnd-jbrun) [inline dataprep-actions/src/test/java/org/talend/dataprep/transformation/actions/math/ModuloTest.java]: Formatting and indentation. Please add use case with wrong parameters like strings that are not numbers. Add test case with a zero division (modulo 0).

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1122/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1122/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1123/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1123/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1131/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1131/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1134/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1134/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1143/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1143/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1148/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1148/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1191/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1191/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1194/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1194/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1208/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1208/)

File: dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages_fr.properties [hunk 8]
```diff
@@ -617,6 +617,17 @@ parameter.mode.desc=Utiliser une valeur ou une autre colonne
 parameter.operand.label=Op\u00E9rateur de comparaison
 parameter.operand.desc=Valeur fixe \u00E0 utiliser pour l''op\u00E9rateur de comparaison
 
+#
+# Modulo
+#
+action.modulo.label=Modulo
+action.modulo.desc=Calculer le modulo résultant d'une division
```
Comment (elogeart) [inline dataprep-actions/src/main/resources/org/talend/dataprep/i18n/actions_messages_fr.properties]: Typo: Missing a `'`

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1209/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1209/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1213/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1213/)
